connector_magento/models/product/common.py

```diff
@@ -108,12 +108,10 @@ def recompute_magento_qty(self):
         informations for each product.
         """
         # group products by backend
-        backends = defaultdict(self.browse)
-        for product in self:
-            backends[product.backend_id] |= product
-
-        for backend, products in backends.iteritems():
-            self._recompute_magento_qty_backend(backend, products)
+        for backend in self.mapped('backend_id'):
+            self._recompute_magento_qty_backend(
+                backend, self.filtered(
+                    lambda x: x.backend_id.id == backend.id))
```

Comment:
But then if you have more than 1 backend, each binding will be iterated N times + 1 where N is the number of backend (1 loop for `filtered` on each backend + 1 loop for `mapped` to obtain the backends). In the  base case when you have 1 backend, you still loop 2 times on the records.
If the recordset's `|=` is a concern (I didn't know it was particularly slow, then I would use a simple set in a dictionary and rebrowse afterwards:

```python
backends = defaultdict(set)
for product in self:
    backends[product.backend_id].add(product.id)

for backend, product_ids in backends.iteritems():
    self._recompute_magento_qty_backend(backend, self.browse(product_ids))